File: design-patterns/python/composite/filesystem/file.py
from typing import List
from .file_system_item import FileSystemItem

class File(FileSystemItem):
    def __init__(self, name: str, size: int):
        self.name = name
        self.size = size

    def get_size(self) -> int:
        return self.size

    def __str__(self) -> str:
        return f"File: {self.name} ({self.size} bytes)" 

File: design-patterns/golang/composite/file.go
package composite

import "fmt"

// File represents a file in the file system
type File struct {
	name string
	size int64
}

// NewFile creates a new File instance
func NewFile(name string, size int64) *File {
	return &File{
		name: name,
		size: size,
	}
}

// GetName returns the name of the file
func (f *File) GetName() string {
	return f.name
}

// GetSize returns the size of the file
func (f *File) GetSize() int64 {
	return f.size
}

// Print prints the file information with the given indentation
func (f *File) Print(indent string) {
	fmt.Printf("%sFile: %s (%d bytes)\n", indent, f.name, f.size)
}

// Delete deletes the file
func (f *File) Delete() {
	fmt.Printf("Deleting file: %s\n", f.name)
}
